### DISCUSSION TITLE
Occuring migration problems for seeded data

### DISCUSSION THREAD

**Discussion by 5m1Ly** [mikro-orm/mikro-orm#4463] [Q&A]
**Describe the bug**
> So within the example where I try to describe the bug best
> 
> We have 2 migrations, and both have a corresponding dataset.
> - migration 1 (initial entity)
> - migration 2 (modified entity)
> 
> We also have 2 datasets (test data that the DB is seeded with)
> - dataset 1 (initial dataset)
> - dataset 2 (modified dataset)

**Migration 1**
So I created a basic entity for which migration 1 is created, then we which to this migration. This is then followed by running the seeder and inserting dataset 1 into the database.

**Migration 2**
At a later point in time, I add the name property to the entity, I don't want this property to be null because each role should have a name also this name should be unique for each role created therefore a default value cant be set. When this is done a new migration is created.

**But when I want to switch to the new migration the problems occur because the first dataset (the one currently in the database) doesn't have the name property and the property is defined to be not null and MySQL or PostgreSQL automatically insert null into a column with missing data.</br></br>This would then be solved by seeding the database again, but this isn't possible because the migration didn't happen due to the SQL error.</br></br>Also if the migration was able to execute the additional data within the second dataset wouldn't update the data that is already there.</br></br> Therefore I need to remove the docker container with the database which is no problem for me when I'm developing but when we ship these changes to production it isn't even a consideration to delete the database.**

#### Migration 1 Entity
``` ts
import { Entity, PrimaryKey } from '@mikro-orm/core';

@Entity({ tableName: 'role' })
export class Role {
    @PrimaryKey({ defaultRaw: 'uuid_generate_v4()', type: 'uuid' })
    id: string;
}
```

#### Dataset 1
``` ts
export const roleDatasets = {
    dataset1: {
        id: 'ee6840d3-232f-476f-a2c2-99034dde1d54',
    },
    dataset2: {
        id: '72c8520e-6ed3-4058-ad70-7e7a34dd9982',
    },
    dataset3: {
        id: '2a35f002-4d5d-4f8c-b650-7a625b002ca4',
    },
};
```

#### Migration 2 Entity
``` ts
import { Entity, PrimaryKey, Property } from '@mikro-orm/core';

@Entity({ tableName: 'role' })
export class Role {
    @PrimaryKey({ defaultRaw: 'uuid_generate_v4()', type: 'uuid' })
    id: string;

    @Property()
    name: string;
}
```

#### Dataset 2
``` ts
export const roleDatasets = {
    dataset1: {
        id: 'ee6840d3-232f-476f-a2c2-99034dde1d54',
        name: 'admin',
    },
    dataset2: {
        id: '72c8520e-6ed3-4058-ad70-7e7a34dd9982',
        name: 'mod',
    },
    dataset3: {
        id: '2a35f002-4d5d-4f8c-b650-7a625b002ca4',
        name: 'user',
    },
};
```

**Versions**

| Dependency | Version |
| - | - |
| node | v18.16.0 |
| typescript | ^4.7.4 |
| @mikro-orm/cli | 5.6.5 |
| @mikro-orm/migrations | 5.6.5 |
| @mikro-orm/seeder | 5.6.5 |
| your-driver | couldn't find the driver within the project code but we use a PostgreSQL db within a docker container and the application is running on WSL2 |

**Comment by B4nan** [comment]
To start with - seeding is generally not something you use on a production database, at least not randomly, regardless of the state - the use case is more about seeding your test database, or feeding the database with initial data - there is no seeding state, e.g. seeding won't care about some seeder being already executed or not, neither about the database state, that is always up to you to handle.

I am not entirely sure what you are reporting here, it sounds more like you are not sure how to operate those things instead of a bug per-se? Sounds like you should be modifying the data in your migrations, not just the schema?
